### PR TITLE
Interval columns

### LINKED ISSUE #15: Support Interval Types
Recent Clickhouse [supports](https://github.com/yandex/ClickHouse/search?l=SQL&q=interval&type=&utf8=%E2%9C%93) intervals / timedeltas but they have special types.

Query:

```sql
SELECT INTERVAL `int_col` DAY AS `tmp`
FROM ibis_testing.`functional_alltypes`
LIMIT 10000
```

Traceback:

```bash    return self.con.execute(query, columnar=True, with_column_types=True)                                                                                                                                                            [2/45772]
../../.pyenv/versions/miniconda3-latest/envs/ml3/lib/python3.6/site-packages/clickhouse_driver/client.py:176: in execute
    types_check=types_check, columnar=columnar
../../.pyenv/versions/miniconda3-latest/envs/ml3/lib/python3.6/site-packages/clickhouse_driver/client.py:227: in process_ordinary_query
    columnar=columnar)
../../.pyenv/versions/miniconda3-latest/envs/ml3/lib/python3.6/site-packages/clickhouse_driver/client.py:75: in receive_result
    self.receive_no_progress_result(result, columnar)
../../.pyenv/versions/miniconda3-latest/envs/ml3/lib/python3.6/site-packages/clickhouse_driver/client.py:102: in receive_no_progress_result
    packet = self.receive_packet()
../../.pyenv/versions/miniconda3-latest/envs/ml3/lib/python3.6/site-packages/clickhouse_driver/client.py:133: in receive_packet
    packet = self.connection.receive_packet()
../../.pyenv/versions/miniconda3-latest/envs/ml3/lib/python3.6/site-packages/clickhouse_driver/connection.py:323: in receive_packet
    packet.block = self.receive_data()
../../.pyenv/versions/miniconda3-latest/envs/ml3/lib/python3.6/site-packages/clickhouse_driver/connection.py:386: in receive_data
    block = self.block_in.read()
../../.pyenv/versions/miniconda3-latest/envs/ml3/lib/python3.6/site-packages/clickhouse_driver/streams/native.py:77: in read
    column = read_column(column_type, n_rows, self.fin)
../../.pyenv/versions/miniconda3-latest/envs/ml3/lib/python3.6/site-packages/clickhouse_driver/columns/service.py:55: in read_column
    column = get_column_by_spec(column_spec)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

spec = 'IntervalDay', column_options = {}

    def get_column_by_spec(spec, column_options=None):
        column_options = column_options or {}

        def create_column_with_options(x):
            return get_column_by_spec(x, column_options)

        if spec.startswith('FixedString'):
            length = int(spec[12:-1])
            return FixedString(length)

        elif spec.startswith('Enum'):
            return create_enum_column(spec)

        elif spec.startswith('Array'):
            return create_array_column(spec, create_column_with_options)

        elif spec.startswith('Nullable'):
            return create_nullable_column(spec, create_column_with_options)

        else:
            try:
                cls = column_by_type[spec]
                return cls(**column_options)

            except KeyError as e:
>               raise errors.UnknownTypeError('Unknown type {}'.format(e.args[0]))
E               clickhouse_driver.errors.UnknownTypeError: Code: 50. Unknown type IntervalDay

../../.pyenv/versions/miniconda3-latest/envs/ml3/lib/python3.6/site-packages/clickhouse_driver/columns/service.py:51: UnknownTypeError
------------------------------------------------- Captured log call --------------------------------------------------
connection.py              174 INFO     Connecting. Database: ibis_testing. User: default
connection.py              275 INFO     Connected to ClickHouse server version 1.1.54310
connection.py              439 INFO     Query: SELECT INTERVAL `int_col` DAY AS `tmp`
FROM ibis_testing.`functional_alltypes`
LIMIT 10000
connection.py              414 DEBUG    Block send time: 8.8e-05
```

### PR DESCRIPTION
**Pull request by kszucs**
Resolves #15

### CODE REVIEW

**Comment by xzkostyan** [comment]
Thanks! I'll check PR later.

**Comment by kszucs** [comment]
Sadly I couldn't setup clickhouse-client locally (OSX) but I guess it'll eventually pass on travis.

**Comment by kszucs** [comment]
@xzkostyan 3.2 build is failing due to missing setuptools-scm, others are passing.

File: tests/columns/test_interval.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+            ('MONTH', 2),
+            ('WEEK', 3),
+            ('DAY', 4),
+            ('HOUR', 5),
+            ('MINUTE', 6),
+            ('SECOND', 7)
+        ]
+        columns = ', '.join(['INTERVAL {} {}'.format(v, k)
+                             for k, v in interval])
+        query = 'SELECT {} FROM system.numbers LIMIT 1'.format(columns)
```
Comment (xzkostyan) [inline tests/columns/test_interval.py]: You can use `'SELECT {}'` as query template. Query `SELECT INTERVAL 1 DAY` works fine without `FROM` and `LIMIT` clauses.

**Comment by kszucs** [comment]
@xzkostyan rebase done

**Comment by xzkostyan** [comment]
Thanks!

**Pull request merged**